src/canvas-extractor/edfi_canvas_extractor/extract_graphql.py

```diff
@@ -165,6 +167,22 @@ def _write_assignments(
     )
 
 
+@catch_exceptions
+def _get_submissions(
+    arguments: MainArguments,
+    gql: GraphQLExtractor,
+    sync_db: sqlalchemy.engine.base.Engine,
+) -> None:
+    logger.info("Extracting Submissions from Canvas")
+    (sections, _, _) = results_store["sections"]
+    logger.info("Writing LMS UDM AssignmentSubmissions to CSV files")
```

Comment:
SHOULD: This can be separated out into it's own function `_write_submissions` with the `@catch_exceptions` decorator like we have for other similar operations in the file.

Reply:
There is a subject here, when its on separate functions stops the mapping. It could be the way that its passing the params to the write function in mapping. Meanwhile I leave it working.